isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/backend/spring.adoc

```diff
@@ -0,0 +1,290 @@
+
+Spring ist ein Applikations-Container, welcher sogenannte Spring-Beans instanziiert, per _Dependency Injection_ konfiguriert und bereitstellt.
+Spring-Beans sind beliebige Java-Klassen.
+Für diese Klassen kann man benötigte andere Spring-Beans oder Konfigurationsparameter konfigurieren, welche der Klasse daraufhin im Konstruktor oder per `set`-Methode übergeben werden.
+
+Konfiguriert werden Spring-Beans und ihre Abhängigkeiten durch von Spring bereitgestellte Annotationen.
+Diese werden beim Start des Applikations-Containers gescannt und ausgewertet (_Component Scan_).
+
+Ein <<image-spring-beans-beispiel-allgemeine-anwendung,Spring-Beans Beispiel>> zeigt einen Ausschnitt der für eine allgemeine Anwendung erstellten Spring-Beans.
+Spring-Beans werden grün dargestellt und besitzen „referenzierte“ Abhängigkeiten zu benötigten anderen Spring-Beans.
+
+[[image-spring-beans-beispiel-allgemeine-anwendung]]
+.Spring-Beans Beispiel einer allgemeinen Anwendung
+image::software-technisch/backend/SpringBeansBeispiel.dn.svg[]
+
+Das Beispiel zeigt bereits, dass nicht alle Klassen der Anwendung als Spring-Beans konfiguriert werden:
+Für die Komponente `Verwaltung` wird eine Klasse als Spring-Bean konfiguriert, welche die Funktionalität der Komponente bereitstellt.
+Generell gilt, dass jede zentrale und wichtige Klasse aber als Spring-Bean konfiguriert werden sollte.
+
+Für die Modellierung und Konfiguration der Spring-Beans werden im folgenden Vorgaben aufgestellt.
+
+[[querschnitt-als-beans-konfigurieren]]
+==== Querschnittliche Funktionalität als Spring-Beans konfigurieren
+
+Querschnittliche Funktionalität (etwa für das Monitoring, für Fehler- und Nachrichtentexte oder für die Versendung von Mails) sind als Spring-Beans zu konfigurieren.
+Ebenfalls über Spring-Beans durchzuführen ist die Konfiguration diverser Frameworks, z.B. Hibernate.
+Die Konfiguration dieser Frameworks wird in ihren Nutzungsvorgaben beschrieben.
+
+[[beans-standardmaessig-als-singletons-definieren]]
+==== Beans standardmäßig als Singletons definieren
+
+Spring-Beans können entweder als Singletons mit nur einer Instanz, mit einer Instanz pro Aufruf oder mit einer Instanz pro Abhängigkeit (Prototype) erzeugt werden.
+Die Spring-Beans einer Anwendung sollen zustandslos sein und werden als Singleton-Beans erzeugt.
+Wo technisch erforderlich können auch andere Scopes verwendet werden.
+
+[[vorgaben-zur-spring-konfiguration]]
+==== Vorgaben zur Spring-Konfiguration
+
+Die Spring-Konfiguration der Anwendung ist nach folgenden Vorgaben zu erstellen.
+
+Der zentrale Ausgangspunkt für die Spring-Konfiguration ist die <<listing-applikationsklasse>>.
+Diese wird im Package der Anwendung (`<org>.<domäne>.<anwendung>`) erstellt.
+Sie ist mit `@Configuration` und `@EnableAutoConfiguration` annotiert.
+Damit wird die Autokonfiguration von Spring Boot eingeschaltet.
+Zusätzlich erbt die Klasse von `SpringBootServletInitializer`, damit die Anwendung im Tomcat deployt werden kann.
```

Comment:
Ist das eine Main-Klasse?
Was ist, wenn wir lokal einen embedded tomcat verwenden?

Reply:
Es ist die zentrale Klasse, die eine Spring-Applikation definiert. Embedded Tomcat ist aktuell noch nicht in Betrachtung.